CMakeTest fails on linux

I get the following error while configuring project on Ubuntu:
```
-- HTTP server failed: 1
/home/.../osslsigncode-2.7/Testing/certs/cert.der: No such file or directory
CMake Error at cmake/CMakeTest.cmake:119 (string):
  string sub-command SUBSTRING requires four arguments.
Call Stack (most recent call first):
  CMakeLists.txt:86 (include)


-- Configuring incomplete, errors occurred!
```

I am guessing `certs` folder should also be copied here https://github.com/mtrojnar/osslsigncode/blob/f9ad19d4a2e1786bd4fccb17240483683769c1b1/cmake/CMakeTest.cmake#L10-L15

I hope this helps in any way
```
$ uname -a
Linux loboda-devlin 4.15.0-142-generic #146~16.04.1-Ubuntu SMP Tue Apr 13 09:27:15 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

I am running custom build on my side, with specific openssl and zlib path
```
/home/loboda/.islcc/cmake-3.25.2-1/bin/cmake -DCMAKE_BUILD_TYPE=Release -DOPENSSL_ROOT_DIR=/home/loboda/.islcc/openssl-3.0.12-2-2848888071450c4fa4a01b54370ae7f6/usr/local/ssl -DZLIB_ROOT=/home/loboda/.islcc/zlib-1.2.8-3-2848888071450c4fa4a01b54370ae7f6 -DCMAKE_INSTALL_PREFIX=/home/loboda/.islcc/osslsigncode-2.7-27f80c2fc39de4598a7b51045c7f2203 .
```
The output I get with this command is
```
- The C compiler identification is GNU 10.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/loboda/Documents/isl/b.ISLConferenceProxyWithModules_linux64.0.main/compiler/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found OpenSSL: /home/loboda/.islcc/openssl-3.0.12-2-2848888071450c4fa4a01b54370ae7f6/usr/local/ssl/lib/libcrypto.a (found version "3.0.12")  
-- Could NOT find CURL (missing: CURL_LIBRARY) (found version "7.47.0")
-- Found ZLIB: /home/loboda/.islcc/zlib-1.2.8-3-2848888071450c4fa4a01b54370ae7f6/lib/libz.a (found version "1.2.8") 
-- Using bash completions dir /home/loboda/.islcc/osslsigncode-2.7-27f80c2fc39de4598a7b51045c7f2203/share/bash-completion/completions
-- Looking for getpass
-- Looking for getpass - found
-- Looking for termios.h
-- Looking for termios.h - found
-- Looking for sys/mman.h
-- Looking for sys/mman.h - found
-- Looking for mmap
-- Looking for mmap - found
-- cURL support disabled (library not found)
-- Found Python3: /usr/bin/python3.5 (found version "3.5.2") found components: Interpreter 
-- HTTP server failed: 1
/home/loboda/.islcc/osslsigncode-2.7-27f80c2fc39de4598a7b51045c7f2203/osslsigncode-2.7/Testing/certs/cert.der: No such file or directory
CMake Error at cmake/CMakeTest.cmake:119 (string):
  string sub-command SUBSTRING requires four arguments.
Call Stack (most recent call first):
  CMakeLists.txt:86 (include)


-- Configuring incomplete, errors occurred!
```

If I add
`"${CMAKE_CURRENT_SOURCE_DIR}/tests/certs"`
to CMakeTest.cmake build is successful.

--- a/cmake/CMakeTest.cmake
+++ b/cmake/CMakeTest.cmake
@@ -70,6 +70,8 @@ else(WIN32 OR APPLE)
             RESULT_VARIABLE server_error)
         if(server_error)
             message(STATUS "HTTP server failed: ${server_error}")
+            message(STATUS "Use python3 to start HTTP server: \"python3 Testing/server_http.py --port 19254\"")
+            set(default_certs 1)
         else(server_error)
             # Check if file exists and is no-empty
             while(NOT EXISTS ${LOGS}/port.log)

--- a/tests/server_http.py
+++ b/tests/server_http.py
@@ -43,6 +43,7 @@ class RequestHandler(SimpleHTTPRequestHandler):
             self.send_response(200)
             self.send_header("Content-type", "application/crl")
             self.end_headers()
+            resp_data = b''
             # Read the file and send the contents
             if url.path == "/intermediateCA":
                 with open(CACRL, 'rb') as file:
@@ -53,6 +54,7 @@ class RequestHandler(SimpleHTTPRequestHandler):
             self.wfile.write(resp_data)
         except Exception as err: # pylint: disable=broad-except
             print(f"HTTP GET request error: {err}")
+
 
     def do_POST(self): # pylint: disable=invalid-name
         """"Serves the POST request type"""
@@ -76,7 +78,7 @@ class RequestHandler(SimpleHTTPRequestHandler):
                 openssl.check_returncode()
                 self.send_header("Content-type", "application/timestamp-reply")
                 self.end_headers()
-                resp_data = None
+                resp_data = b''
                 with open(RESPONS, mode="rb") as file:
                     resp_data = file.read()
                 self.wfile.write(resp_data)

--- a/tests/server_http.py
+++ b/tests/server_http.py
@@ -53,7 +53,7 @@ class RequestHandler(SimpleHTTPRequestHandler):
                     resp_data = file.read()
             self.wfile.write(resp_data)
         except Exception as err: # pylint: disable=broad-except
-            print(f"HTTP GET request error: {err}")
+            print("HTTP GET request error: {}".format(err))
 
 
     def do_POST(self): # pylint: disable=invalid-name
@@ -83,7 +83,7 @@ class RequestHandler(SimpleHTTPRequestHandler):
                     resp_data = file.read()
                 self.wfile.write(resp_data)
         except Exception as err: # pylint: disable=broad-except
-            print(f"HTTP POST request error: {err}")
+            print("HTTP POST request error: {}".format(err))
 
 
 class HttpServerThread():
@@ -100,7 +100,7 @@ class HttpServerThread():
         self.server_thread = threading.Thread(target=self.server.serve_forever)
         self.server_thread.start()
         hostname, port = self.server.server_address[:2]
-        print(f"HTTP server started, URL http://{hostname}:{port}")
+        print("HTTP server started, URL http://{}:{}".format(hostname, port))
         return port
 
 
@@ -121,7 +121,7 @@ def main() -> None:
         with open(PORT_LOG, mode="w") as file:
             file.write("{}".format(port))
     except OSError as err:
-        print(f"OSError: {err}")
+        print("OSError: {}".format(err))
         ret = err.errno
     finally:
         sys.exit(ret)
@@ -133,7 +133,7 @@ if __name__ == '__main__':
         if fpid > 0:
             sys.exit(0)
     except OSError as ferr:
-        print(f"Fork #1 failed: {ferr.errno} {ferr.strerror}")
+        print("Fork #1 failed: {} {}".format(ferr.errno, ferr.strerror))
         sys.exit(1)
 
     try:
@@ -141,7 +141,7 @@ if __name__ == '__main__':
         if fpid > 0:
             sys.exit(0)
     except OSError as ferr:
-        print(f"Fork #2 failed: {ferr.errno} {ferr.strerror}")
+        print("Fork #2 failed: {} {}".format(ferr.errno, ferr.strerror))
         sys.exit(1)
 
     # Start the daemon main loop

--- a/tests/server_http.pyw
+++ b/tests/server_http.pyw
@@ -43,6 +43,7 @@ class RequestHandler(SimpleHTTPRequestHandler):
             self.send_response(200)
             self.send_header("Content-type", "application/crl")
             self.end_headers()
+            resp_data = b''
             # Read the file and send the contents
             if url.path == "/intermediateCA":
                 with open(CACRL, 'rb') as file:
@@ -53,6 +54,7 @@ class RequestHandler(SimpleHTTPRequestHandler):
             self.wfile.write(resp_data)
         except Exception as err: # pylint: disable=broad-except
             print(f"HTTP GET request error: {err}")
+
 
     def do_POST(self): # pylint: disable=invalid-name
         """"Serves the POST request type"""
@@ -76,7 +78,7 @@ class RequestHandler(SimpleHTTPRequestHandler):
                 openssl.check_returncode()
                 self.send_header("Content-type", "application/timestamp-reply")
                 self.end_headers()
-                resp_data = None
+                resp_data = b''
                 with open(RESPONS, mode="rb") as file:
                     resp_data = file.read()
                 self.wfile.write(resp_data)

--- a/tests/server_http.pyw
+++ b/tests/server_http.pyw
@@ -53,7 +53,7 @@ class RequestHandler(SimpleHTTPRequestHandler):
                     resp_data = file.read()
             self.wfile.write(resp_data)
         except Exception as err: # pylint: disable=broad-except
-            print(f"HTTP GET request error: {err}")
+            print("HTTP GET request error: {}".format(err))
 
 
     def do_POST(self): # pylint: disable=invalid-name
@@ -83,7 +83,7 @@ class RequestHandler(SimpleHTTPRequestHandler):
                     resp_data = file.read()
                 self.wfile.write(resp_data)
         except Exception as err: # pylint: disable=broad-except
-            print(f"HTTP POST request error: {err}")
+            print("HTTP POST request error: {}".format(err))
 
 
 class HttpServerThread():
@@ -100,7 +100,7 @@ class HttpServerThread():
         self.server_thread = threading.Thread(target=self.server.serve_forever)
         self.server_thread.start()
         hostname, port = self.server.server_address[:2]
-        print(f"HTTP server started, URL http://{hostname}:{port}")
+        print("HTTP server started, URL http://{}:{}".format(hostname, port))
         return port
 
 
@@ -115,7 +115,7 @@ def main() -> None:
         with open(PORT_LOG, mode="w") as file:
             file.write("{}".format(port))
     except OSError as err:
-        print(f"OSError: {err}")
+        print("OSError: {}".format(err))
         ret = err.errno
     finally:
         sys.exit(ret)

